### PR TITLE
syrupy 4

### PR DESCRIPTION
**Pull request by alangenfeld**
using `uv` in `tox` started to resolve to the new version of syrupy which has api changes and new snapshot formats.

Instead of pinning back I opted to migrate forward. 

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19984?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#19871** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19871?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19984** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19984?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#19983** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19983?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19958** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19958?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19984?utm_source=stack-comment).

File: python_modules/libraries/dagster-polars/setup.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def get_version() -> str:
         "deltalake": ["deltalake>=0.15.0"],
         "gcp": ["dagster-gcp>=0.19.5"],
         "test": [
-            "pytest>=7.3.1",
```
Comment (smackesey) [inline python_modules/libraries/dagster-polars/setup.py:48]: is this supposed to be here

**Review by smackesey — APPROVED** [review]
👍

File: python_modules/libraries/dagster-polars/setup.py (hunk 1 above)
Comment (alangenfeld) [reply python_modules/libraries/dagster-polars/setup.py:48]: it failed before i moved it here https://buildkite.com/dagster/dagster-dagster/builds/76501#018dd221-c4ba-4258-aefb-834405b9802e

**Pull request merged**